### ISSUE TITLE
Differences with torchmetrics.image.StructuralSimilarityIndexMeasure?

### ISSUE THREAD

**Issue by oliver-batchelor** [rahul-goel/fused-ssim#1]
This is great - the ssim is a huge component in the GS loss.

I tested it out and there's a small difference between fused-ssim and StructuralSimilarityIndexMeasure from torchmetrics.image, 
I can't figure out why - however when training with torchmetrics SSIM I get about ~ 1ish PSNR better on most scenes. As far as I can tell the parameters are the same, sigma=1.5 on both... any clues?

In terms of the ssim value it's always slightly higher for fused-ssim (in all the situations I have observed anyway). (Edit - this difference goes away if I don't torch.compile the torchmetrics SSIM)

**Comment by oliver-batchelor** [comment]
Seems like when I replace SSIM with StructuralSimilarityIndexMeasure in the tests, it matches for the forward pass but not the gradient.

**Comment by rahul-goel** [comment]
Hi Oliver.

I quickly tested with `torchmetrics.image.StructuralSimilarityIndexMeasure` and it is passing the tests with the `valid` padding configuration described in the README file:

```
ssim_value = fused_ssim(predicted_image, gt_image, padding="valid")
```

You can test this by replacing [this line](https://github.com/rahul-goel/fused-ssim/blob/84422e0da94c516220eb3acedb907e68809e9e01/tests/test.py#L60C7-L60C47) with `pm_ssim = StructuralSimilarityIndexMeasure(data_range=1.0).cuda()`.

Let me know if this works.

**Comment by oliver-batchelor** [comment]
Interesting—this is the configuration I was testing. I think perhaps I messed something up, as when I reverted, it tested fine. Sorry about that!

I found the discrepancy in training; it stems from a mistake swapping the gt/predicted arguments, where the pytorch SSIM(s) are symmetrical and fused-ssim is not when it comes to the gradient. 

After fixing this it behaves exactly the same. Thanks again for this work, it really is very useful!

Oliver
